addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -199,6 +199,198 @@ def delete_bill(self, bill_id):
         except Exception as e:
             return "Failed to delete bill: {}".format(e)
 
+    ##create/get/delete_bills payment
+    @api.model
+    def create_bill_payment(self, bill_ids, journal_id, payment_date, payment_method_line_id):
+        AccountMove = self.env['account.move']
+        bills = AccountMove.browse(bill_ids)
+
+        if not bills.exists():
+            raise UserError("No valid bills found with the provided IDs.")
+        if any(bill.state != 'posted' for bill in bills):
+            raise UserError("All bills must be posted to proceed with payment.")
+
+        payment_register = self.env['account.payment.register'].with_context({
+            'active_model': 'account.move',
+            'active_ids': bill_ids,
+        }).create({
+            'payment_date': payment_date,
+            'journal_id': journal_id,
+            'payment_method_line_id': payment_method_line_id,
+        })
+
+        payment_register.action_create_payments()
+
+        payment_data_list = []  # Initialize an empty list to store payment data
+
+        for bill in bills:
+            # Adjust the lambda to the correct field and value for 'payable' account types
+            account_move_lines_to_reconcile = bill.line_ids.filtered(
+                lambda line: not line.reconciled and line.account_id.account_type == 'payable'
+            )
+
+            # Search for move lines linked to this bill that are not yet reconciled
+            move_lines_to_reconcile = self.env['account.move.line'].search([
+                ('move_id', 'in', bill_ids),
+                ('reconciled', '=', True),
+            ])
+
+            # Payments are linked to move lines, so you can reconcile them through move lines
+            if move_lines_to_reconcile:
+
+                for move_line in move_lines_to_reconcile:
+                    payment = move_line.payment_id
```

Comment:
This payment_id is not being added in the returned data. Can you add this to the returned data. We need to test if the id of the created bill payment is successfully returned as a non-zero value